test/framework/seedregistrationframework.go

```diff
@@ -0,0 +1,386 @@
+	flag.StringVar(&newCfg.ShootedSeedNamespace, "shoot-namespace", "", "Namespace of the shoot")
+	flag.StringVar(&newCfg.ShootedSeedKubeconfig, "shoot-kubecfg", "", "Path to the kubeconfig of the shooted seed")
+	// Provider
+	flag.StringVar(&newCfg.ProviderType, "provider-type", "", "Provider type e.g. aws, az, gcp")
+	flag.StringVar(&newCfg.ProviderRegion, "provider-region", "", "Provider region e.g. eu-west-1")
+	// Network
+	flag.StringVar(&newCfg.NetworkingBlockCIDRs, "networking-blockcidrs", "", "Comma-separated values for seed.Spec.Networks.BlockCIDRs.")
+	flag.StringVar(&newCfg.NetworkingPods, "networking-pods", "", "Value for Seed.Spec.Networks.Pods")
+	flag.StringVar(&newCfg.NetworkingServices, "networking-services", "", "Value for Seed.Spec.Networks.Services")
+	flag.StringVar(&newCfg.NetworkingNodes, "networking-nodes", "", "Value for Seed.Spec.Networks.Nodes")
+	flag.StringVar(&newCfg.DeployGardenletOnShoot, "deploy-gardenlet", "", "Specify if a gardanlet instance should be installed on the shoot. Default = true")
+
+	seedRegistrationConfig = newCfg
+
+	return seedRegistrationConfig
+}
+
+// AddSeed function tries to get an existing seed using the configured GardenClient
+// and the given seed parameter and adds it to the SeedRegistrationFramework
+func (f *SeedRegistrationFramework) AddSeed(ctx context.Context, seed *gardencorev1beta1.Seed) error {
+	if f.GardenClient == nil {
+		return errors.New("no gardener client is defined")
+	}
+	if err := f.GardenClient.DirectClient().Get(ctx, client.ObjectKey{Name: seed.Name}, seed); err != nil {
+		return err
+	}
+	f.Seed = seed
+	return nil
+}
+
+// RegisterSeed registers a new seed and its secret reference (seed.Spec.SecretRef)
+// according to the configuration provided for the SeedRegistrationFramework
+func (f *SeedRegistrationFramework) RegisterSeed(ctx context.Context) (err error) {
+	if f.GardenClient == nil {
+		return errors.New("no gardener client is defined")
+	}
+
+	refShoot := &gardencorev1beta1.Shoot{}
+	if err = f.GardenClient.DirectClient().Get(ctx, kutil.Key(f.Config.ShootedSeedNamespace, f.Config.ShootedSeedName), refShoot); err != nil {
+		return err
+	}
+
+	if deploy, _ := strconv.ParseBool(f.Config.DeployGardenletOnShoot); deploy {
+		return f.annotateShootToBeUsedAsSeed(ctx, refShoot)
+	}
+
+	secretBinding := &gardencorev1beta1.SecretBinding{}
+	if err = f.GardenClient.DirectClient().Get(ctx, kutil.Key(f.Config.ShootedSeedNamespace, f.Config.SecretBinding), secretBinding); err != nil {
+		return err
+	}
+
+	f.Seed = f.buildSeedObject(secretBinding, refShoot)
+
+	if err = f.buildSeedSecret(ctx, secretBinding); err != nil {
+		f.Logger.Errorf("Cannot build seed secret %s", err)
+		return err
+	}
+	f.Logger.Infof("Creating seed secret %s in namespace %s", f.Secret.GetName(), f.Secret.GetNamespace())
+	// Apply secret to the cluster
+	if err = client.IgnoreNotFound(f.GardenClient.DirectClient().Create(ctx, f.Secret)); err != nil {
```

Comment:
To my knowledge there is no way for `.Create` to return a not found error. 

```suggestion
	if err = f.GardenClient.DirectClient().Create(ctx, f.Secret); err != nil {
```